fsspec_reference_maker/hdf.py

```diff
@@ -93,13 +135,20 @@ def transfer_attrs(self, h5obj, zobj):
             try:
                 zobj.attrs[n] = v
             except TypeError:
-                print(f'Caught TypeError: {n}@{h5obj.name} = {v} ({type(v)})')
+                lggr.exception(
+                    f'Caught TypeError: {n}@{h5obj.name} = {v} ({type(v)})')
 
-    def translator(self, name, h5obj):
+    def translator(self, name: str, h5obj: Union[h5py.Dataset, h5py.Group]):
         """Produce Zarr metadata for all groups and datasets in the HDF5 file.
         """
         if isinstance(h5obj, h5py.Dataset):
-            lggr.debug(f'Dataset: {h5obj.name}')
+            lggr.debug(f'HDF5 dataset: {h5obj.name}')
+            if h5obj.id.get_create_plist().get_layout() == h5py.h5d.COMPACT:
+                RuntimeError(
+                    f'Compact HDF5 datasets not yet supported: <{h5obj.name} '
```

Comment:
Please create an issue referencing this exception and your comment about this in the other issue. Hopefully we can cope with this soon.